tfx/components/transform/executor.py

```diff
@@ -350,27 +360,38 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
 
     analyze_data_paths = []
     for split in splits_config.analyze:
-      data_uri = artifact_utils.get_split_uri(input_dict[EXAMPLES_KEY], split)
-      analyze_data_paths.append(io_utils.all_files_pattern(data_uri))
+      data_uris = artifact_utils.get_split_uris(input_dict[EXAMPLES_KEY], split)
+      if len(data_uris) != len(input_dict[EXAMPLES_KEY]):
+        raise ValueError(
+            'Analyze split does not exist over all example artifacts: %s' %
+            split)
+      for data_uri in data_uris:
+        analyze_data_paths.append(io_utils.all_files_pattern(data_uri))
 
     transform_data_paths = []
     materialize_output_paths = []
     if output_dict.get(TRANSFORMED_EXAMPLES_KEY) is not None:
-      transformed_example_artifact = artifact_utils.get_single_instance(
-          output_dict[TRANSFORMED_EXAMPLES_KEY])
-      # TODO(b/161490287): move the split_names setting to executor for all
-      # components.
-      transformed_example_artifact.split_names = (
-          artifact_utils.encode_split_names(list(splits_config.transform)))
+      for transformed_example_artifact in output_dict[TRANSFORMED_EXAMPLES_KEY]:
+        # TODO(b/161490287): move the split_names setting to executor for all
+        # components.
+        transformed_example_artifact.split_names = (
+            artifact_utils.encode_split_names(list(splits_config.transform)))
 
       for split in splits_config.transform:
-        data_uri = artifact_utils.get_split_uri(input_dict[EXAMPLES_KEY], split)
-        transform_data_paths.append(io_utils.all_files_pattern(data_uri))
-        transformed_example = os.path.join(
-            artifact_utils.get_split_uri(output_dict[TRANSFORMED_EXAMPLES_KEY],
-                                         split),
-            _DEFAULT_TRANSFORMED_EXAMPLES_PREFIX)
-        materialize_output_paths.append(transformed_example)
+        data_uris = artifact_utils.get_split_uris(input_dict[EXAMPLES_KEY],
+                                                  split)
+        if len(data_uris) != len(input_dict[EXAMPLES_KEY]):
+          raise ValueError(
+              'Transform split does not exist over all example artifacts: %s' %
+              split)
```

Comment:
maybe this should be in the get_split_uris function?

Reply:
updated util function to do the length check.